modxcms/revolution
# Big trash refactor

### What does it do?

1. Adds new "Trash" menu item to the top menu
2. Fixes #14190
3. Fixes trash button icon color change when its inactive
4. Removes unnecessary styles
5. Adds a purge button to the edit resource page
6. The resource tree and trash button state are now updates when a resource is deleted/undeleted on the resource update page
7. Fixes trash button tooltip information when purging resources on the trash page
8. Now when a resource have been delete/undelete in the resource tree, the state of the resource update form (checkbox and buttons) updates
9. Fixes undefined text of the trash button tooltip
10. Adds purge resource menu item to the context menu in the resource tree

### Why is it needed?

There are many errors when deleting/restoring/purging resources

### How to test

1. —
2. Click by the middle mouse button on the trash button on the toolbar
3. Hover over the trash button when its inactive, before the icon changed color, now when inactive it does not
4. —
5. Open the edit resource page, delete it and you will see a new purging button
6. If you delete or restore a resource on the edit page, the resource tree and the trash button in the toolbar remain unchanged. Now the resource tree and the trash button will change their state
7. On the trash page when purging resources, the state of the trash button in the toolbar was not updated, now it will be
8. Create a resource, open its edit page and delete the resource **in the resource tree**, the state of the checkbox and buttons in the edit form will not change. Now everything works as it should.
9. If you delete a resource and when you hover the trash button, it will be undefined in the tooltip. Now the information will be correct
10. Open the context menu of resource tree, delete the resource and open again

### Related issue(s)/PR(s)
#14190
